module/vision/BallDetector/src/BallDetector.cpp

```diff
@@ -276,20 +307,25 @@ namespace module::vision {
                         fmt::format("Distance Throwout {}", std::abs(projection_distance - distance) / max_distance));
                     log<NUClear::DEBUG>("**************************************************");
 
+                    // If this ball didn't pass the checks, don't keep it
                     if (!keep) {
                         b.measurements.clear();
                     }
-                    if (log_level <= NUClear::DEBUG || keep) {
+                    // If the ball passed the checks, add it to the Balls message to be emitted
+                    // If it didn't pass the checks, but we're debugging, then emit the ball to see throwouts in NUsight
+                    if (keep || log_level <= NUClear::DEBUG) {
                         balls->balls.push_back(std::move(b));
                     }
                 }
 
+                // Emit the Balls message, even if no balls were found
                 emit(std::move(balls));
             });
     }
 }  // namespace module::vision
 
 
+// Can we delete the below code?
```

Comment:
What's the go with this code, can it be removed? Should I make an issue about adding a check in? Or is the degree of fit check above a similar thing?

Reply:
This code was intended to be used rather than averaging the edge points to get the centre and radius but according to @Bidski it didn't work so well.

That said averaging has it's problems too so it's possible this could still be used and improved to work properly.

Reply:
I'll have a look at this too then. There's NUbots and NUWebots branches for vision webots tests for ball accuracy so once that's done can properly improve and test if it's an improvement.